arch/arm/src/stm32/stm32_uart.h

```diff
@@ -538,16 +538,31 @@
 #  define HAVE_RS485 1
 #endif
 
-#ifdef HAVE_RS485
-#  define USART_CR1_USED_INTS    (USART_CR1_RXNEIE | USART_CR1_TXEIE | USART_CR1_PEIE | USART_CR1_TCIE)
+#if defined(HAVE_RS485)
+#  define USART_CR1_USED_INTS \
+         (USART_CR1_RXNEIE  | USART_CR1_TXEIE   | \
+          USART_CR1_PEIE    | USART_CR1_TCIE    | \
+          USART_CR1_IDLEIE)
 #else
-#  define USART_CR1_USED_INTS    (USART_CR1_RXNEIE | USART_CR1_TXEIE | USART_CR1_PEIE)
+#  define USART_CR1_USED_INTS \
+         (USART_CR1_RXNEIE  | USART_CR1_TXEIE   | \
+          USART_CR1_PEIE    | USART_CR1_IDLEIE)
 #endif
 
+#define USART_CR3_USED_INTS \
+       (USART_CR3_EIE)
```

Comment:
Is CR3 on all the serial IP types of the SoCs? I.E. Not on an F1 but on F3 - etc

Reply:
I've checked following SoCs: F0x0, F0x1, F0x4, F0x8, F100, F1x1, F1x2, F1x3, F1x5, F1x7, F301, F334, F37x, F405, F407, F401, F446. They all have all 3 control registers. G4xx have compatibility with F3. Skipped a few, If anything comes to mind, note it and I can check.